### PR TITLE
Share value ordering

### PR DESCRIPTION
**Pull request by tysen**
Corporations that end up on the same stock market space after a series
of sales by the same player should retain the relative operating order
they had before the sales.

### CODE REVIEW

File: lib/engine/share_price.rb [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 module Engine
   class SharePrice
-    attr_reader :coordinates, :price, :corporations, :can_par, :type, :types
+    attr_reader :coordinates, :price, :can_par, :type, :types
```
Comment (tobymao) [inline lib/engine/share_price.rb:5]: i don't this change is necessary, you can just mutate corporations

corporations.clear
corporations.concat(reordered)

File: lib/engine/step/g_1849/buy_sell_par_shares.rb [hunk 2]
```diff
@@ -6,6 +6,23 @@ module Engine
...
     module G1849
       class BuySellParShares < BuySellParShares
+        def setup
+          super
+          @game.old_operating_order = @game.corporations.sort
+        end
+
+        def pass!
+          @passed = true
+          if @current_actions.any?
```
Comment (tobymao) [inline lib/engine/step/g_1849/buy_sell_par_shares.rb:16]: unless empty

File: lib/engine/share_price.rb [hunk 3]
```diff
@@ -59,6 +60,10 @@ def initialize(coordinates,
       @limited = !unlimited_types.include?(type)
     end
 
+    def ==(other)
```
Comment (tobymao) [inline lib/engine/share_price.rb:63]: why did you add this?

File: lib/engine/share_price.rb (hunk 1 above)
Comment (tysen) [reply lib/engine/share_price.rb:5]: good point

File: lib/engine/share_price.rb (hunk 3 above)
Comment (tysen) [reply lib/engine/share_price.rb:63]: For the `group_by` in `reorder_corps`

**Review by tobymao — APPROVED** [review]
